### PR TITLE
YANG validation for ConfigDB Updates: MIRROR_SESSION use case

### PR DESCRIPTION
**Pull request by isabelmsft**
#### What I did
-Fix ValidatedConfigDBConnector decorator code to account for adding to a table that is possibly nonexistent to begin with
-Add YANG validation via ValidatedConfigDBConnector decorator for MIRROR_SESSION use case
#### How I did it
-Used GCU within ValidatedConfigDBConnector decorator for YANG validation

#### How to verify it
Verified locally on virtual switch

### CODE REVIEW

**Comment by isabelmsft** [comment]
Blocking YANG model issue: https://github.com/sonic-net/sonic-buildimage/issues/12397

File: config/validated_config_db_connector.py
Comment (qiluo-msft) [inline config/validated_config_db_connector.py:16]: >is_table_present_config_db [](http://example.com/codeflow?start=4&length=26)

this implementation is not efficient.

File: config/validated_config_db_connector.py
Comment (qiluo-msft) [inline config/validated_config_db_connector.py:36]: >is_table_present_config_db [](http://example.com/codeflow?start=27&length=26)

Need to check key or table existence.

File: config/validated_config_db_connector.py
Comment (isabelmsft) [reply config/validated_config_db_connector.py:16]: Made implementation more efficient in this PR- https://github.com/sonic-net/sonic-utilities/pull/2452

After PR 2452 is merged, I will rebase this MIRROR_SESSION PR to include those changes because this MIRROR_SESSION PR is blocked by YANG issue https://github.com/sonic-net/sonic-buildimage/issues/12397 , and PR 2452 will likely be merged first

File: config/validated_config_db_connector.py
Comment (isabelmsft) [reply config/validated_config_db_connector.py:36]: Also addressed this comment in PR https://github.com/sonic-net/sonic-utilities/pull/2452

**Comment by zhangyanzhao** [comment]
@qiluo-msft  can you please review this PR? Thanks.

File: config/validated_config_db_connector.py
Comment (qiluo-msft) [reply config/validated_config_db_connector.py:16]: 2452 is merged. Could you rebase this one?

File: config/validated_config_db_connector.py
Comment (qiluo-msft) [reply config/validated_config_db_connector.py:36]: I see it resolved in 2452.

**Review by qiluo-msft — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**
